core/services/wifi/main.py

```diff
@@ -62,7 +62,14 @@ async def saved() -> Any:
 @version(1, 0)
 async def connect(credentials: WifiCredentials) -> Any:
     try:
-        await wifi_manager.set_wifi_password(credentials)
+        saved_networks = await wifi_manager.get_saved_wifi_network()
+        for network in saved_networks:
```

Comment:
The code could be simpler with the inverse logic:
```
network = find(lambda network: network.ssid == credentials.ssid, networks)
if not network:
    network_id = await wifi_manager.add_network(credentials)
else:
    network_id = network.id
```

Reply:
Definitely better. Changing.

Reply:
Changed.